### PR TITLE
Weather

### PR DESCRIPTION
**Pull request by JZasinets**

### CODE REVIEW

File: mentoring-react/src/components/Weather/form.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+const Form = ({ getWeather }) => {
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/form.js:3]: memo

File: mentoring-react/src/components/Weather/form.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    const refreshResults = () => {
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/form.js:4]: useCallback

File: mentoring-react/src/components/Weather/weather.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import DisplayWeather from './displayWeather';
+import Form from './form';
+
+class Weather extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            city: undefined,
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/weather.js:9]: null is default value

File: mentoring-react/src/components/Weather/weather.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    }
+
+    render () {
+        return (
+            <div className='weather'>
+                <div className='header'>Прогноз погоды</div>
+                <Form
+                    getWeather={this.getWeather}
+                    city={this.state.city}
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/weather.js:62]: Destructuring from this.state

File: mentoring-react/src/components/navbar.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+const Navbar = () => {
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/navbar.js:3]: memo

File: mentoring-react/src/components/Weather/form.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        return getWeather;
+    }
+
+    return (
+        <>
+            <div className='weather-text'>Введите название города и код страны на английском:</div>
+            <form className='weather-form' onSubmit={getWeather}>
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/form.js:11]:
```suggestion
            <form className='weather-form' onSubmit={refreshResults}>
```

File: mentoring-react/src/components/Weather/displayWeather.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+const DisplayWeather = React.memo(({city, code, description, temp, minimumTemp, maximumTemp, error}) => {
+    const valueTemperature = (value) => {
+        return Math.round(value - 273);
+    };
+
+    return (
+        <>
+            { city && code ?
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/displayWeather.js:10]:
```suggestion
            { Boolean(city && code) ?
```

File: mentoring-react/src/components/Weather/displayWeather.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+const DisplayWeather = React.memo(({city, code, description, temp, minimumTemp, maximumTemp, error}) => {
+    const valueTemperature = (value) => {
```
Comment (ragozin-nikita) [inline mentoring-react/src/components/Weather/displayWeather.js:4]: useCallback

File: mentoring-react/package.json [hunk 9]
```diff
@@ -34,5 +34,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (ragozin-nikita) [inline mentoring-react/package.json:38]: dependency instead of devDependencies

**Review by ragozin-nikita — CHANGES_REQUESTED** [review]

File: mentoring-react/src/components/Weather/form.js (hunk 1 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/form.js]: done

File: mentoring-react/src/components/Weather/form.js (hunk 2 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/form.js]: done

File: mentoring-react/src/components/Weather/weather.js (hunk 3 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/weather.js]: done

File: mentoring-react/src/components/navbar.js (hunk 5 above)
Comment (JZasinets) [reply mentoring-react/src/components/navbar.js]: done

File: mentoring-react/src/components/Weather/form.js (hunk 6 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/form.js]: done

File: mentoring-react/src/components/Weather/displayWeather.js (hunk 7 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/displayWeather.js]: done

File: mentoring-react/src/components/Weather/displayWeather.js (hunk 8 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/displayWeather.js]: done

File: mentoring-react/package.json (hunk 9 above)
Comment (JZasinets) [reply mentoring-react/package.json]: Done

File: mentoring-react/src/components/Weather/weather.js (hunk 4 above)
Comment (JZasinets) [reply mentoring-react/src/components/Weather/weather.js]: done

**Pull request merged**